fdbserver/ClusterRecovery.actor.cpp

```diff
@@ -1186,7 +1187,14 @@ ACTOR Future<Void> readTransactionSystemState(Reference<ClusterRecoveryData> sel
 	// Recover version info
 	self->lastEpochEnd = oldLogSystem->getEnd() - 1;
 	if (self->lastEpochEnd == 0) {
-		self->recoveryTransactionVersion = 1;
+		// Set the initial cluster version equal to the current timestamp in
+		// microseconds, relative to a defined epoch. Versions advance at
+		// roughtly one million versions per second, so one version equals one
+		// microsecond.
+		self->recoveryTransactionVersion = g_network->timer() * SERVER_KNOBS->VERSIONS_PER_SECOND;
+		if (!g_network->isSimulated()) {
+			self->recoveryTransactionVersion -= SERVER_KNOBS->DEFAULT_VERSION_EPOCH * SERVER_KNOBS->VERSIONS_PER_SECOND;
+		}
```

Comment:
An option here would be to choose your current time deterministically. For example, you could choose a random number of seconds since the reference time as your start time.

I'm not sure if this would end up posing any problems for tooling, since it's possible some tools rely on the 0 start time.